pkg/asset/installconfig/aws/basedomain.go

```diff
@@ -52,26 +55,47 @@ func GetBaseDomain() (string, error) {
 	}
 	sort.Strings(publicZones)
 	if len(publicZones) == 0 {
-		return "", errors.New("no public Route 53 hosted zones found")
+		return "", "", errors.New("no public Route 53 hosted zones found")
 	}
 
-	var domain string
 	if err := survey.AskOne(&survey.Select{
 		Message: "Base Domain",
 		Help:    "The base domain of the cluster. All DNS records will be sub-domains of this base and will also include the cluster name.\n\nIf you don't see you intended base-domain listed, create a new public Route53 hosted zone and rerun the installer.",
 		Options: publicZones,
-	}, &domain, func(ans interface{}) error {
+	}, &baseDomain, func(ans interface{}) error {
 		choice := ans.(string)
 		i := sort.SearchStrings(publicZones, choice)
 		if i == len(publicZones) || publicZones[i] != choice {
 			return errors.Errorf("invalid base domain %q", choice)
 		}
 		return nil
 	}); err != nil {
-		return "", errors.Wrap(err, "failed UserInput for base domain")
+		return "", "", errors.Wrap(err, "failed UserInput for base domain")
 	}
 
-	return domain, nil
+	zoneList, ok := publicZoneMap[baseDomain]
+	if !ok {
+		return "", "", errors.Errorf("invalid base domain %q", baseDomain)
+	}
+
+	zoneListLen := len(zoneList)
+	if zoneListLen <= 0 {
```

Comment:
hmm, when do you think zoneListLen is going to be 0 or -ve ?

Reply:
I will remove this entire check. It is curious that len() does return a signed integer though.

Reply:
In looking at this further, I feel it is okay for the zero length check. Why would you not check for a zero length list?